bridge/discord.go

```diff
@@ -223,7 +223,15 @@ func (d *discordBot) publishReaction(s *discordgo.Session, r *discordgo.MessageR
 			// HACK: theoretically could already be there, thereotically not a big problem
 			originalMessage.Mentions = append(originalMessage.Mentions, originalMessage.Author)
 		}
-		originalMessage.Content = fmt.Sprintf(" to <%s> %s", username, TruncateString(40, originalMessage.Content))
+		originalMessage.Content = fmt.Sprintf(
+			" to <%s> %s",
+			username,
+			// Truncate messages to just 40 characters so reactions to long messages
+			// don't pollute the IRC log. Similarly, replace newlines with spaces
+			// so that any reactions to messages with a newline within the first 40
+			// characters don't cause multiple IRC messages to be sent.
+			strings.ReplaceAll(TruncateString(40, originalMessage.Content), "\n", " "),
```

Comment:
could you strings.ReplaceAll the inner string first, so that we don't count newlines when truncating?

Reply:
Given we're replacing it with space, wouldn't the count be identical? `TruncateString()` seems to use `IsSpace` internally (which would treat both newline and space the same) so there shouldn't be any changes functionally.

Reply:
oh, true